### PR TITLE
:white_check_mark: Add unit tests to package build

### PR DESCRIPTION
**Pull request by MaliaLabor**
:bug: Updated return type for i2c_minimum_speed::transaction

### CODE REVIEW

**Review by kammce — CHANGES_REQUESTED** [review]
Can you update the CI so it runs correctly. Currently it still points to libhal/libhal when it should point to libhal/ci

**Comment by MaliaLabor** [comment]
> Can you update the CI so it runs correctly. Currently it still points to libhal/libhal when it should point to libhal/ci

Done

File: include/libhal-soft/i2c_minimum_speed.hpp [hunk 1]
```diff
@@ -57,13 +57,14 @@ class minimum_speed_i2c : public hal::i2c
...
-  status driver_transaction(
+  result<transaction_t> driver_transaction(
     hal::byte p_address,
     std::span<const hal::byte> p_data_out,
     std::span<hal::byte> p_data_in,
     hal::function_ref<hal::timeout_function> p_timeout) override
   {
-    return m_i2c->transaction(p_address, p_data_out, p_data_in, p_timeout);
+    return HAL_CHECK(
+      m_i2c->transaction(p_address, p_data_out, p_data_in, p_timeout));
```
Comment (kammce) [inline include/libhal-soft/i2c_minimum_speed.hpp]: Actually... this should be possible without `HAL_CHECK`.

**Review by kammce — CHANGES_REQUESTED** [review]

File: include/libhal-soft/i2c_minimum_speed.hpp (hunk 1 above)
Comment (MaliaLabor) [reply include/libhal-soft/i2c_minimum_speed.hpp]: Ok, I changed it back

**Review by kammce — APPROVED** [review]

**Pull request merged**